### ISSUE TITLE
Building without the minify parameter can lead to subsequent builds failing

### ISSUE THREAD

**Issue by Oglesby** [lodash/lodash#256]
When the minify parameter is given, the build callback isn't called until the closure compiler and uglifyjs are fully downloaded and the minify process completes. This works fine.

When the parameter isn't given, lodash is built without minify, the build callback is invoked, then the minify process is started. The minify process then begins downloading the closure compiler. The code that started the lodash build continues on its merry way (because the build callback has been invoked) then exits, aborting the download. 

The next time the build is run, lodash is built, the callback is invoked, then the minify process is started. The minify process sees that the closure compiler has already been downloaded (sorta), so starts downloading uglifyjs. Uglifyjs is small enough that it completes before the calling process exits. The actual minify starts, which tries to use the bad closure compiler jar and errors.

Basically, the minify process should only be started and the closure compiler and uglifyjs downloaded if the minify parameter is given.
